### ISSUE TITLE
Add the machine BNL modex which is important for FATES 

### ISSUE THREAD

**Issue by ekluzek** [ESMCI/cime#2077]
We need to add the BNL machine modex for FATES work. @rgknox. Here are the changes in cime5.3.0-alpha.02:

```
diff --git a/cime/config/cesm/machines/config_batch.xml b/cime/config/cesm/machines/config_batch.xml
index 07ad1e39..cbbd9735 100644
--- a/cime/config/cesm/machines/config_batch.xml
+++ b/cime/config/cesm/machines/config_batch.xml
@@ -460,6 +460,17 @@
     </queues>
   </batch_system>
 
+  <!-- modex is PBS -->
+  <batch_system MACH="modex" type="pbs">
+    <directives>
+      <directive>-l nodes={{ num_nodes }}:ppn={{ tasks_per_node }}</directive>
+      <directive default="/bin/bash" > -S {{ shell }}  </directive>
+    </directives>
+    <queues>
+      <queue walltimemax="36:00:00" default="true">batch</queue>
+    </queues>
+  </batch_system>
+
   <batch_jobs>
     <!-- order matters, with no-batch jobs will be run in the order listed here -->
     <job name="case.run">
diff --git a/cime/config/cesm/machines/config_compilers.xml b/cime/config/cesm/machines/config_compilers.xml
index f20871ea..ab4f22db 100644
--- a/cime/config/cesm/machines/config_compilers.xml
+++ b/cime/config/cesm/machines/config_compilers.xml
@@ -1098,4 +1098,15 @@ using a fortran linker.
   <SFC MPILIB="mpi-serial">gfortran</SFC>
 </compiler>
 
+<compiler COMPILER="gnu" MACH="modex">
+  <NETCDF_PATH>$ENV{NETCDF_HOME}</NETCDF_PATH>
+  <SLIBS>
+    <append> -L$ENV{HDF5_HOME}/lib -lhdf5_fortran -lhdf5 -lhdf5_hl -lhdf5hl_fortran </append>
+    <append> -L$ENV{NETCDF_HOME}/lib/ -lnetcdff -lnetcdf -lcurl -lblas -llapack</append>
+  </SLIBS>
+  <CPPDEFS>
+     <append MODEL="gptl"> -DHAVE_VPRINTF -DHAVE_GETTIMEOFDAY -DHAVE_BACKTRACE </append>
+  </CPPDEFS>
+</compiler>
+
 </config_compilers>
diff --git a/cime/config/cesm/machines/config_machines.xml b/cime/config/cesm/machines/config_machines.xml
index 0b72a031..00391fcd 100644
--- a/cime/config/cesm/machines/config_machines.xml
+++ b/cime/config/cesm/machines/config_machines.xml
@@ -1856,6 +1856,69 @@
       <env name="MP_MPILIB">$MPILIB</env>
     </environment_variables>
   </machine>
+  
+  <machine MACH="modex">
+      <DESC>Medium sized linux cluster at BNL, torque scheduler.</DESC>
+      <OS>LINUX</OS>
+      <COMPILERS>gnu</COMPILERS>
+      <MPILIBS>openmpi,mpi-serial</MPILIBS>
+      <CIME_OUTPUT_ROOT>/data/$ENV{USER}</CIME_OUTPUT_ROOT>
+      <DIN_LOC_ROOT>/data/Model_Data/cesm_input_datasets/</DIN_LOC_ROOT>
+      <DIN_LOC_ROOT_CLMFORC>/data/Model_Data/cesm_input_datasets/atm/datm7</DIN_LOC_ROOT_CLMFORC>
+      <DOUT_S_ROOT>$CIME_OUTPUT_ROOT/cesm_archive/$CASE</DOUT_S_ROOT>
+      <DOUT_L_MSROOT>csm/$CASE</DOUT_L_MSROOT>
+      <BASELINE_ROOT>$CIME_OUTPUT_ROOT/cesm_baselines</BASELINE_ROOT>
+      <CCSM_CPRNC>/$CIME_OUTPUT_ROOT/cesm_tools/cprnc/cprnc</CCSM_CPRNC>
+      <GMAKE_J>4</GMAKE_J>
+      <BATCH_SYSTEM>pbs</BATCH_SYSTEM>
+      <SUPPORTED_BY>rgknox at lbl dot gov and sserbin at bnl gov</SUPPORTED_BY>
+      <MAX_TASKS_PER_NODE>12</MAX_TASKS_PER_NODE>
+      <PES_PER_NODE>12</PES_PER_NODE>
+      <PROJECT_REQUIRED>FALSE</PROJECT_REQUIRED>
+      <mpirun mpilib="mpi-serial">
+          <executable></executable>
+      </mpirun>
+      <mpirun mpilib="default">
+          <executable>mpirun</executable>
+          <arguments>
+              <arg name="num_tasks">-np $TOTALPES</arg>
+              <arg name="tasks_per_node">-npernode $PES_PER_NODE</arg>
+          </arguments>
+      </mpirun>
+      <module_system type="module">
+          <init_path lang="sh">/etc/profile.d/modules.sh</init_path>
+          <init_path lang="csh">/etc/profile.d/modules.csh</init_path>
+          <init_path lang="perl">/usr/share/Modules/init/perl.pm</init_path>
+          <init_path lang="python">/usr/share/Modules/init/python.py</init_path>
+          <cmd_path lang="sh">module</cmd_path>
+          <cmd_path lang="csh">module</cmd_path>
+          <cmd_path lang="perl">/usr/bin/modulecmd perl</cmd_path>
+          <cmd_path lang="python">/usr/bin/modulecmd python</cmd_path>
+          <modules>
+              <command name="purge"/>
+              <command name="load">perl/5.22.1</command>
+              <command name="load">libxml2/2.9.2</command>
+              <command name="load">maui/3.3.1</command>
+              <command name="load">python/2.7.13</command>
+          </modules>
+          <modules compiler="gnu">
+              <command name="load">gcc/5.4.0</command>
+              <command name="load">gfortran/5.4.0</command>
+              <command name="load">hdf5/1.8.19fates</command>
+              <command name="load">netcdf/4.4.1.1-gnu540-fates</command>
+              <command name="load">openmpi/2.1.1-gnu540</command>
+          </modules>
+          <modules compiler="gnu" mpilib="!mpi-serial">
+              <command name="load">openmpi/2.1.1-gnu540</command>
+          </modules>
+      </module_system>
+       <environment_variables>
+         <env name="HDF5_HOME">/data/software/hdf5/1.8.19fates</env>
+         <env name="NETCDF_HOME">/data/software/netcdf/4.4.1.1-gnu540-fates</env>
+       </environment_variables>
+
+  </machine>
+
 <default_run_suffix>
   <default_run_exe>${EXEROOT}/cesm.exe </default_run_exe>
   <default_run_misc_suffix> >> cesm.log.$LID 2>&amp;1 </default_run_misc_suffix>
diff --git a/cime/config/cesm/machines/config_pio.xml b/cime/config/cesm/machines/config_pio.xml
index ae0faedd..6c436999 100644
--- a/cime/config/cesm/machines/config_pio.xml
+++ b/cime/config/cesm/machines/config_pio.xml
@@ -52,6 +52,7 @@
       <value mach="hobart" compiler="nag">netcdf</value>
       <value mach="lbl-lr3">netcdf</value>
       <value mach="lbl-lr2">netcdf</value>
+      <value mach="modex">netcdf</value>
       <value mpilib="mpi-serial">netcdf</value>
       <value compset="CLM45%ED">netcdf</value>
     </values>
```

**Comment by rgknox** [comment]
@ekluzek, this may be another issue, but with our older version of CLM, we aren't able to use pnetcdf with CLM50ED.

<value compset="CLM50%ED">netcdf</value>

**Comment by rgknox** [comment]
I'm also noticing a difference in how directives are specified in PBS
a la modex:

```<directive>-l nodes={{ num_nodes }}:ppn={{ tasks_per_node }}</directive>```

a la cheyenne:

```<directive> -l select={{ num_nodes }}:ncpus={{ max_tasks_per_node }}:mpiprocs={{ tasks_per_node }}:ompthreads={{ thread_count }}</directive> ```

Its unclear to me if the differences have to do with updates to how cime interprets these arguments, or updates in the PBS software that is on cheyenne, versus older builds.

**Comment by ekluzek** [comment]
@rgknox I'm assuming the netcdf issue is resolved in cime master, since we are able to run with fates. We had the same problem. So I left that change off. I did look on master and don't see that change. So it might be good for you to try a FATES simulation with the latest CLM tag (which has a later cime than fates-clm) and verify it works.

**Comment by ekluzek** [comment]
@rgknox on the PBS differences, yes those are changes that are due to the cime update. We had the same issue with adding the lbl-lr machines, but it was resolved in the scripts testing. So I figure the same correction will happen when we create the PR for this as well. In the end we'll want you to test on these machines when we have a CLM with a newer cime and these changes in place.

**Comment by serbinsh** [comment]
@ekluzek I am watching this but dont have permission to add myself to this issue.

**Comment by ekluzek** [comment]
@serbinsh OK, it looks like you aren't a member of the ESMCI group. Which is maybe fine if you won't normally be doing changes in ESMCI. Let us know if you think you should be a member. 

But, it's also good to have you both posting and mentioned here.

**Comment by serbinsh** [comment]
@ekluzek Sure, please add me to the group.

**Comment by rljacob** [comment]
Anyone with a github account can post to this open repo and be @ mentioned.  He would need to be added to get assigned an issue or write to the repo.

**Comment by serbinsh** [comment]
@ekluzek @rljacob good point. I am watching this repo now and that may be sufficient.  However if we add the modex cluster to cime I may need to push changes to the machine configs later on so I may need to be able to request changes anyway. 

@ekluzek FOr now, I have a clean version of CLM and am ready to try with the updated CIME when you have a version of the PR ready for test on modex so just let me know

**Comment by rgknox** [comment]
Thanks @ekluzek.  Shawn and I just had a pow-wow and we are readying ourselves to test when called.  Just to confirm: it sounds like you are going to do another pass, and we should hold before we start tinkering on modex?  Or was it the other way around?

**Comment by ekluzek** [comment]
@rgknox and @serbinsh. Yes, I'll do another pass and create a branch and PR for you to test. I'll assign both of you to review as well as Jim Edwards from our side.
